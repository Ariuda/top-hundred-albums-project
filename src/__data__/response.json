{
    "feed": {
        "entry": [
            {
                "id": {
                    "attributes": {
                        "im:id": "999999"
                    }
                },
                "im:name": {
                    "label": "Album title"
                }, 
                "im:artist": { 
                    "label": "Album author", 
                    "attributes": {
                        "href": "link"
                    }
                },
                "im:image": [
                    {
                        "label": "image-string"
                    },
                    {
                        "label": "image-string"
                    },
                    {
                        "label": "image-string"
                    }
                ],
                "im:price": {
                    "label": "album-price"
                },
                "category": {
                    "attributes": {
                        "label": "category",
                        "scheme": "category-link"
                    }
                }
            },
            {
                "id": {
                    "attributes": {
                        "im:id": "9999998"
                    }
                },
                "im:name": {
                    "label": "Album title"
                }, 
                "im:artist": { 
                    "label": "Album author", 
                    "attributes": {
                        "href": "link"
                    }
                },
                "im:image": [
                    {
                        "label": "image-string"
                    },
                    {
                        "label": "image-string"
                    },
                    {
                        "label": "image-string"
                    }
                ],
                "im:price": {
                    "label": "album-price"
                },
                "category": {
                    "attributes": {
                        "label": "category",
                        "scheme": "category-link"
                    }
                }
            },
            {
                "id": {
                    "attributes": {
                        "im:id": "9999997"
                    }
                },
                "im:name": {
                    "label": "Album title"
                }, 
                "im:artist": { 
                    "label": "Album author", 
                    "attributes": {
                        "href": "link"
                    }
                },
                "im:image": [
                    {
                        "label": "image-string"
                    },
                    {
                        "label": "image-string"
                    },
                    {
                        "label": "image-string"
                    }
                ],
                "im:price": {
                    "label": "album-price"
                },
                "category": {
                    "attributes": {
                        "label": "category",
                        "scheme": "category-link"
                    }
                }
            },
            {
                "id": {
                    "attributes": {
                        "im:id": "9999996"
                    }
                },
                "im:name": {
                    "label": "Album title"
                }, 
                "im:artist": { 
                    "label": "Album author", 
                    "attributes": {
                        "href": "link"
                    }
                },
                "im:image": [
                    {
                        "label": "image-string"
                    },
                    {
                        "label": "image-string"
                    },
                    {
                        "label": "image-string"
                    }
                ],
                "im:price": {
                    "label": "album-price"
                },
                "category": {
                    "attributes": {
                        "label": "category",
                        "scheme": "category-link"
                    }
                }
            },
            {
                "id": {
                    "attributes": {
                        "im:id": "9999995"
                    }
                },
                "im:name": {
                    "label": "Album title"
                }, 
                "im:artist": { 
                    "label": "Album author", 
                    "attributes": {
                        "href": "link"
                    }
                },
                "im:image": [
                    {
                        "label": "image-string"
                    },
                    {
                        "label": "image-string"
                    },
                    {
                        "label": "image-string"
                    }
                ],
                "im:price": {
                    "label": "album-price"
                },
                "category": {
                    "attributes": {
                        "label": "category",
                        "scheme": "category-link"
                    }
                }
            },
            {
                "id": {
                    "attributes": {
                        "im:id": "9999994"
                    }
                },
                "im:name": {
                    "label": "Album title"
                }, 
                "im:artist": { 
                    "label": "Album author", 
                    "attributes": {
                        "href": "link"
                    }
                },
                "im:image": [
                    {
                        "label": "image-string"
                    },
                    {
                        "label": "image-string"
                    },
                    {
                        "label": "image-string"
                    }
                ],
                "im:price": {
                    "label": "album-price"
                },
                "category": {
                    "attributes": {
                        "label": "category",
                        "scheme": "category-link"
                    }
                }
            },
            {
                "id": {
                    "attributes": {
                        "im:id": "9999993"
                    }
                },
                "im:name": {
                    "label": "Album title"
                }, 
                "im:artist": { 
                    "label": "Album author", 
                    "attributes": {
                        "href": "link"
                    }
                },
                "im:image": [
                    {
                        "label": "image-string"
                    },
                    {
                        "label": "image-string"
                    },
                    {
                        "label": "image-string"
                    }
                ],
                "im:price": {
                    "label": "album-price"
                },
                "category": {
                    "attributes": {
                        "label": "category",
                        "scheme": "category-link"
                    }
                }
            },
            {
                "id": {
                    "attributes": {
                        "im:id": "9999992"
                    }
                },
                "im:name": {
                    "label": "Album title"
                }, 
                "im:artist": { 
                    "label": "Album author", 
                    "attributes": {
                        "href": "link"
                    }
                },
                "im:image": [
                    {
                        "label": "image-string"
                    },
                    {
                        "label": "image-string"
                    },
                    {
                        "label": "image-string"
                    }
                ],
                "im:price": {
                    "label": "album-price"
                },
                "category": {
                    "attributes": {
                        "label": "category",
                        "scheme": "category-link"
                    }
                }
            },
            {
                "id": {
                    "attributes": {
                        "im:id": "9999991"
                    }
                },
                "im:name": {
                    "label": "Album title"
                }, 
                "im:artist": { 
                    "label": "Album author", 
                    "attributes": {
                        "href": "link"
                    }
                },
                "im:image": [
                    {
                        "label": "image-string"
                    },
                    {
                        "label": "image-string"
                    },
                    {
                        "label": "image-string"
                    }
                ],
                "im:price": {
                    "label": "album-price"
                },
                "category": {
                    "attributes": {
                        "label": "category",
                        "scheme": "category-link"
                    }
                }
            },
            {
                "id": {
                    "attributes": {
                        "im:id": "9999990"
                    }
                },
                "im:name": {
                    "label": "Album title"
                }, 
                "im:artist": { 
                    "label": "Album author", 
                    "attributes": {
                        "href": "link"
                    }
                },
                "im:image": [
                    {
                        "label": "image-string"
                    },
                    {
                        "label": "image-string"
                    },
                    {
                        "label": "image-string"
                    }
                ],
                "im:price": {
                    "label": "album-price"
                },
                "category": {
                    "attributes": {
                        "label": "category",
                        "scheme": "category-link"
                    }
                }
            },
            {
                "id": {
                    "attributes": {
                        "im:id": "9999989"
                    }
                },
                "im:name": {
                    "label": "Album title"
                }, 
                "im:artist": { 
                    "label": "Album author", 
                    "attributes": {
                        "href": "link"
                    }
                },
                "im:image": [
                    {
                        "label": "image-string"
                    },
                    {
                        "label": "image-string"
                    },
                    {
                        "label": "image-string"
                    }
                ],
                "im:price": {
                    "label": "album-price"
                },
                "category": {
                    "attributes": {
                        "label": "category",
                        "scheme": "category-link"
                    }
                }
            },
            {
                "id": {
                    "attributes": {
                        "im:id": "9999988"
                    }
                },
                "im:name": {
                    "label": "Album title"
                }, 
                "im:artist": { 
                    "label": "Album author", 
                    "attributes": {
                        "href": "link"
                    }
                },
                "im:image": [
                    {
                        "label": "image-string"
                    },
                    {
                        "label": "image-string"
                    },
                    {
                        "label": "image-string"
                    }
                ],
                "im:price": {
                    "label": "album-price"
                },
                "category": {
                    "attributes": {
                        "label": "category",
                        "scheme": "category-link"
                    }
                }
            },
            {
                "id": {
                    "attributes": {
                        "im:id": "9999987"
                    }
                },
                "im:name": {
                    "label": "Album title"
                }, 
                "im:artist": { 
                    "label": "Album author", 
                    "attributes": {
                        "href": "link"
                    }
                },
                "im:image": [
                    {
                        "label": "image-string"
                    },
                    {
                        "label": "image-string"
                    },
                    {
                        "label": "image-string"
                    }
                ],
                "im:price": {
                    "label": "album-price"
                },
                "category": {
                    "attributes": {
                        "label": "category",
                        "scheme": "category-link"
                    }
                }
            },
            {
                "id": {
                    "attributes": {
                        "im:id": "9999986"
                    }
                },
                "im:name": {
                    "label": "Album title"
                }, 
                "im:artist": { 
                    "label": "Album author", 
                    "attributes": {
                        "href": "link"
                    }
                },
                "im:image": [
                    {
                        "label": "image-string"
                    },
                    {
                        "label": "image-string"
                    },
                    {
                        "label": "image-string"
                    }
                ],
                "im:price": {
                    "label": "album-price"
                },
                "category": {
                    "attributes": {
                        "label": "category",
                        "scheme": "category-link"
                    }
                }
            },
            {
                "id": {
                    "attributes": {
                        "im:id": "9999985"
                    }
                },
                "im:name": {
                    "label": "Album title"
                }, 
                "im:artist": { 
                    "label": "Album author", 
                    "attributes": {
                        "href": "link"
                    }
                },
                "im:image": [
                    {
                        "label": "image-string"
                    },
                    {
                        "label": "image-string"
                    },
                    {
                        "label": "image-string"
                    }
                ],
                "im:price": {
                    "label": "album-price"
                },
                "category": {
                    "attributes": {
                        "label": "category",
                        "scheme": "category-link"
                    }
                }
            },
            {
                "id": {
                    "attributes": {
                        "im:id": "9999984"
                    }
                },
                "im:name": {
                    "label": "Album title"
                }, 
                "im:artist": { 
                    "label": "Album author", 
                    "attributes": {
                        "href": "link"
                    }
                },
                "im:image": [
                    {
                        "label": "image-string"
                    },
                    {
                        "label": "image-string"
                    },
                    {
                        "label": "image-string"
                    }
                ],
                "im:price": {
                    "label": "album-price"
                },
                "category": {
                    "attributes": {
                        "label": "category",
                        "scheme": "category-link"
                    }
                }
            },
            {
                "id": {
                    "attributes": {
                        "im:id": "9999983"
                    }
                },
                "im:name": {
                    "label": "Album title"
                }, 
                "im:artist": { 
                    "label": "Album author", 
                    "attributes": {
                        "href": "link"
                    }
                },
                "im:image": [
                    {
                        "label": "image-string"
                    },
                    {
                        "label": "image-string"
                    },
                    {
                        "label": "image-string"
                    }
                ],
                "im:price": {
                    "label": "album-price"
                },
                "category": {
                    "attributes": {
                        "label": "category",
                        "scheme": "category-link"
                    }
                }
            },
            {
                "id": {
                    "attributes": {
                        "im:id": "9999982"
                    }
                },
                "im:name": {
                    "label": "Album title"
                }, 
                "im:artist": { 
                    "label": "Album author", 
                    "attributes": {
                        "href": "link"
                    }
                },
                "im:image": [
                    {
                        "label": "image-string"
                    },
                    {
                        "label": "image-string"
                    },
                    {
                        "label": "image-string"
                    }
                ],
                "im:price": {
                    "label": "album-price"
                },
                "category": {
                    "attributes": {
                        "label": "category",
                        "scheme": "category-link"
                    }
                }
            },
            {
                "id": {
                    "attributes": {
                        "im:id": "9999981"
                    }
                },
                "im:name": {
                    "label": "Album title"
                }, 
                "im:artist": { 
                    "label": "Album author", 
                    "attributes": {
                        "href": "link"
                    }
                },
                "im:image": [
                    {
                        "label": "image-string"
                    },
                    {
                        "label": "image-string"
                    },
                    {
                        "label": "image-string"
                    }
                ],
                "im:price": {
                    "label": "album-price"
                },
                "category": {
                    "attributes": {
                        "label": "category",
                        "scheme": "category-link"
                    }
                }
            },
            {
                "id": {
                    "attributes": {
                        "im:id": "9999980"
                    }
                },
                "im:name": {
                    "label": "Album title"
                }, 
                "im:artist": { 
                    "label": "Album author", 
                    "attributes": {
                        "href": "link"
                    }
                },
                "im:image": [
                    {
                        "label": "image-string"
                    },
                    {
                        "label": "image-string"
                    },
                    {
                        "label": "image-string"
                    }
                ],
                "im:price": {
                    "label": "album-price"
                },
                "category": {
                    "attributes": {
                        "label": "category",
                        "scheme": "category-link"
                    }
                }
            },
            {
                "id": {
                    "attributes": {
                        "im:id": "9999979"
                    }
                },
                "im:name": {
                    "label": "Album title"
                }, 
                "im:artist": { 
                    "label": "Album author", 
                    "attributes": {
                        "href": "link"
                    }
                },
                "im:image": [
                    {
                        "label": "image-string"
                    },
                    {
                        "label": "image-string"
                    },
                    {
                        "label": "image-string"
                    }
                ],
                "im:price": {
                    "label": "album-price"
                },
                "category": {
                    "attributes": {
                        "label": "category",
                        "scheme": "category-link"
                    }
                }
            },
            {
                "id": {
                    "attributes": {
                        "im:id": "9999978"
                    }
                },
                "im:name": {
                    "label": "Album title"
                }, 
                "im:artist": { 
                    "label": "Album author", 
                    "attributes": {
                        "href": "link"
                    }
                },
                "im:image": [
                    {
                        "label": "image-string"
                    },
                    {
                        "label": "image-string"
                    },
                    {
                        "label": "image-string"
                    }
                ],
                "im:price": {
                    "label": "album-price"
                },
                "category": {
                    "attributes": {
                        "label": "category",
                        "scheme": "category-link"
                    }
                }
            },
            {
                "id": {
                    "attributes": {
                        "im:id": "9999977"
                    }
                },
                "im:name": {
                    "label": "Album title"
                }, 
                "im:artist": { 
                    "label": "Album author", 
                    "attributes": {
                        "href": "link"
                    }
                },
                "im:image": [
                    {
                        "label": "image-string"
                    },
                    {
                        "label": "image-string"
                    },
                    {
                        "label": "image-string"
                    }
                ],
                "im:price": {
                    "label": "album-price"
                },
                "category": {
                    "attributes": {
                        "label": "category",
                        "scheme": "category-link"
                    }
                }
            },
            {
                "id": {
                    "attributes": {
                        "im:id": "9999976"
                    }
                },
                "im:name": {
                    "label": "Album title"
                }, 
                "im:artist": { 
                    "label": "Album author", 
                    "attributes": {
                        "href": "link"
                    }
                },
                "im:image": [
                    {
                        "label": "image-string"
                    },
                    {
                        "label": "image-string"
                    },
                    {
                        "label": "image-string"
                    }
                ],
                "im:price": {
                    "label": "album-price"
                },
                "category": {
                    "attributes": {
                        "label": "category",
                        "scheme": "category-link"
                    }
                }
            },
            {
                "id": {
                    "attributes": {
                        "im:id": "9999975"
                    }
                },
                "im:name": {
                    "label": "Album title"
                }, 
                "im:artist": { 
                    "label": "Album author", 
                    "attributes": {
                        "href": "link"
                    }
                },
                "im:image": [
                    {
                        "label": "image-string"
                    },
                    {
                        "label": "image-string"
                    },
                    {
                        "label": "image-string"
                    }
                ],
                "im:price": {
                    "label": "album-price"
                },
                "category": {
                    "attributes": {
                        "label": "category",
                        "scheme": "category-link"
                    }
                }
            },
            {
                "id": {
                    "attributes": {
                        "im:id": "9999974"
                    }
                },
                "im:name": {
                    "label": "Album title"
                }, 
                "im:artist": { 
                    "label": "Album author", 
                    "attributes": {
                        "href": "link"
                    }
                },
                "im:image": [
                    {
                        "label": "image-string"
                    },
                    {
                        "label": "image-string"
                    },
                    {
                        "label": "image-string"
                    }
                ],
                "im:price": {
                    "label": "album-price"
                },
                "category": {
                    "attributes": {
                        "label": "category",
                        "scheme": "category-link"
                    }
                }
            },
            {
                "id": {
                    "attributes": {
                        "im:id": "9999973"
                    }
                },
                "im:name": {
                    "label": "Album title"
                }, 
                "im:artist": { 
                    "label": "Album author", 
                    "attributes": {
                        "href": "link"
                    }
                },
                "im:image": [
                    {
                        "label": "image-string"
                    },
                    {
                        "label": "image-string"
                    },
                    {
                        "label": "image-string"
                    }
                ],
                "im:price": {
                    "label": "album-price"
                },
                "category": {
                    "attributes": {
                        "label": "category",
                        "scheme": "category-link"
                    }
                }
            },
            {
                "id": {
                    "attributes": {
                        "im:id": "9999972"
                    }
                },
                "im:name": {
                    "label": "Album title"
                }, 
                "im:artist": { 
                    "label": "Album author", 
                    "attributes": {
                        "href": "link"
                    }
                },
                "im:image": [
                    {
                        "label": "image-string"
                    },
                    {
                        "label": "image-string"
                    },
                    {
                        "label": "image-string"
                    }
                ],
                "im:price": {
                    "label": "album-price"
                },
                "category": {
                    "attributes": {
                        "label": "category",
                        "scheme": "category-link"
                    }
                }
            }
        ] 
    }
}
