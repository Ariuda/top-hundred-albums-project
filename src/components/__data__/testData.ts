export const testData = [
    {
        id: '123',
        link: 'link',
        title: 'title 1',
        artist: {
            name: 'name 1',
            link: 'link-one',
        },
        image: 'image-one',
        price: 'price-one',
        category: {
            name: 'category',
            link: 'category-link'
        },
        numberOfItems: '8',
        releaseDate: 'release-date',
        rights: 'rights-data',
        index: 1,
        favorite: false  
    },
    {
        id: '124',
        link: 'link',
        title: 'title 2',
        artist: {
            name: 'name 2',
            link: 'link',
        },
        image: 'image',
        price: 'price-two',
        category: {
            name: 'category',
            link: 'category-link'
        },
        numberOfItems: '8',
        releaseDate: 'release-date',
        rights: 'rights-data',
        index: 2,
        favorite: false   
    }
];