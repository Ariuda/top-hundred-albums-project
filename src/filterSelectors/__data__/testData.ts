export const testData = [
    {
        id: '123',
        link: 'link',
        title: 'title 1',
        artist: {
            name: 'name 1',
            link: 'link-one',
        },
        image: 'image-one',
        price: 'price-one',
        category: {
            name: 'category',
            link: 'category-link'
        },
        numberOfItems: '8',
        releaseDate: 'release-date',
        rights: 'rights-data',
        index: 1,
        favorite: false  
    },
    {
        id: '124',
        link: 'link',
        title: 'title 2',
        artist: {
            name: 'name 2',
            link: 'link',
        },
        image: 'image',
        price: 'price-two',
        category: {
            name: 'category',
            link: 'category-link'
        },
        numberOfItems: '8',
        releaseDate: 'release-date',
        rights: 'rights-data',
        index: 2,
        favorite: false   
    },
    {
        id: '125',
        link: 'link',
        title: 'title 3',
        artist: {
            name: 'name 3',
            link: 'link',
        },
        image: 'image',
        price: 'price',
        category: {
            name: 'category',
            link: 'category-link'
        },
        numberOfItems: '8',
        releaseDate: 'release-date',
        rights: 'rights-data',
        index: 3,
        favorite: false   
    },
    {
        id: '126',
        link: 'link',
        title: 'title 4',
        artist: {
            name: 'name 4',
            link: 'link',
        },
        image: 'image',
        price: 'price',
        category: {
            name: 'category',
            link: 'category-link'
        },
        numberOfItems: '8',
        releaseDate: 'release-date',
        rights: 'rights-data',
        index: 4,
        favorite: false   
    },
    {
        id: '126',
        link: 'link',
        title: 'title 5',
        artist: {
            name: 'name 5',
            link: 'link',
        },
        image: 'image',
        price: 'price',
        category: {
            name: 'category',
            link: 'category-link'
        },
        numberOfItems: '8',
        releaseDate: 'release-date',
        rights: 'rights-data',
        index: 5,
        favorite: false   
    },
    {
        id: '127',
        link: 'link',
        title: 'title 6',
        artist: {
            name: 'name 6',
            link: 'link',
        },
        image: 'image',
        price: 'price',
        category: {
            name: 'category',
            link: 'category-link'
        },
        numberOfItems: '8',
        releaseDate: 'release-date',
        rights: 'rights-data',
        index: 6,
        favorite: false   
    },
    {
        id: '128',
        link: 'link',
        title: 'title 7',
        artist: {
            name: 'name 7',
            link: 'link',
        },
        image: 'image',
        price: 'price',
        category: {
            name: 'category',
            link: 'category-link'
        },
        numberOfItems: '8',
        releaseDate: 'release-date',
        rights: 'rights-data',
        index: 7,
        favorite: false   
    },
    {
        id: '129',
        link: 'link',
        title: 'title 8',
        artist: {
            name: 'name 8',
            link: 'link',
        },
        image: 'image',
        price: 'price',
        category: {
            name: 'category',
            link: 'category-link'
        },
        numberOfItems: '8',
        releaseDate: 'release-date',
        rights: 'rights-data',
        index: 8,
        favorite: false   
    },
    {
        id: '130',
        link: 'link',
        title: 'title 9',
        artist: {
            name: 'name 9',
            link: 'link',
        },
        image: 'image',
        price: 'price',
        category: {
            name: 'category',
            link: 'category-link'
        },
        numberOfItems: '8',
        releaseDate: 'release-date',
        rights: 'rights-data',
        index: 9,
        favorite: false   
    },
    {
        id: '131',
        link: 'link',
        title: 'title 10',
        artist: {
            name: 'name 10',
            link: 'link',
        },
        image: 'image',
        price: 'price',
        category: {
            name: 'category',
            link: 'category-link'
        },
        numberOfItems: '8',
        releaseDate: 'release-date',
        rights: 'rights-data',
        index: 10,
        favorite: false   
    },
    {
        id: '132',
        link: 'link',
        title: 'title 11',
        artist: {
            name: 'name 11',
            link: 'link',
        },
        image: 'image',
        price: 'price',
        category: {
            name: 'category',
            link: 'category-link'
        },
        numberOfItems: '8',
        releaseDate: 'release-date',
        rights: 'rights-data',
        index: 11,
        favorite: false   
    },
    {
        id: '133',
        link: 'link',
        title: 'title 12',
        artist: {
            name: 'name 12',
            link: 'link',
        },
        image: 'image',
        price: 'price',
        category: {
            name: 'category',
            link: 'category-link'
        },
        numberOfItems: '8',
        releaseDate: 'release-date',
        rights: 'rights-data',
        index: 12,
        favorite: false   
    },
    {
        id: '134',
        link: 'link',
        title: 'title 13',
        artist: {
            name: 'name 13',
            link: 'link',
        },
        image: 'image',
        price: 'price',
        category: {
            name: 'category',
            link: 'category-link'
        },
        numberOfItems: '8',
        releaseDate: 'release-date',
        rights: 'rights-data',
        index: 13,
        favorite: false   
    },
    {
        id: '135',
        link: 'link',
        title: 'album 14',
        artist: {
            name: 'name 14',
            link: 'link',
        },
        image: 'image',
        price: 'price',
        category: {
            name: 'category',
            link: 'category-link'
        },
        numberOfItems: '8',
        releaseDate: 'release-date',
        rights: 'rights-data',
        index: 14,
        favorite: false   
    },
    {
        id: '136',
        link: 'link',
        title: 'different title 15',
        artist: {
            name: 'name 15',
            link: 'link',
        },
        image: 'image',
        price: 'price',
        category: {
            name: 'category',
            link: 'category-link'
        },
        numberOfItems: '8',
        releaseDate: 'release-date',
        rights: 'rights-data',
        index: 15,
        favorite: false   
    }
];