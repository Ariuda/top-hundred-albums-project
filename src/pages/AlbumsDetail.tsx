import React from 'react';
import { connect } from 'react-redux';

import { RootState } from '../store/reducers';
import { Album } from '../entities';
import { RouteComponentProps } from 'react-router';
import NotFound from './NotFound';

interface Props {
    selectedAlbum?: Album
}

interface MatchParams {
    id: string;
}

type AlbumsDetailProps = Props & MatchParams & RouteComponentProps<MatchParams>;

const AlbumsDetail: React.FC<AlbumsDetailProps> = (props) => {
    const album = props.selectedAlbum;
    if (album) {
        return (
            <div className="albums-detail-wrapper">   
                <span className="go-back" onClick={() => props.history.goBack()}>Go Back</span>
                <div className="albums-detail">
                    <img src={album.image} alt={album.title} />
                    <div className="text-container">
                        <h1>{album.title}</h1>
                        <h2>By {album.artist.name}</h2>
                        <p>Number {album.index} on iTunes</p>
                        <p>Released in {album.releaseDate}</p>
                        <p>{album.numberOfItems} songs</p>
                        <p>Price in iTunes: {album.price}</p>
                        <p>{album.rights}</p> 
                    </div>
                </div>
            </div>
        )
    }
    return <NotFound {...props}/>;
       
};

const mapStateToProps = (state: RootState, props: RouteComponentProps<MatchParams>):Props => {
    return {
        selectedAlbum: state.albumsState.albums.find(album => album.id === props.match.params.id)
    }
}

export default connect(mapStateToProps, null)(AlbumsDetail);