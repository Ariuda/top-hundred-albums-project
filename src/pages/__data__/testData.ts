import { Album } from "../../entities";

export const testData: Album = {
        id: '123',
        link: 'link',
        title: 'title 1',
        artist: {
            name: 'name 1',
            link: 'link-one',
        },
        image: 'image-one',
        price: 'price-one',
        category: {
            name: 'category',
            link: 'category-link'
        },
        numberOfItems: '8',
        releaseDate: 'release-date',
        rights: 'rights-data',
        index: 1,
        favorite: false  
    };