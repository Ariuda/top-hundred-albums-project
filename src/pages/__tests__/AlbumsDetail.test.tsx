import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import { AlbumsDetail } from '../AlbumsDetail';
import { testData } from '../__data__/testData';

const props = {
    selectedAlbum: testData,
    history: {} as any,
    location: {} as any,
    match: {} as any,
};

it('should render the albums detail page correctly', () => {
    const wrapper = shallow(<AlbumsDetail {...props} />);
    expect(toJSON(wrapper)).toMatchSnapshot();
});