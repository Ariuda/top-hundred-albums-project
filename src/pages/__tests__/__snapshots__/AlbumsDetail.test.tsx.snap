// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the albums detail page correctly 1`] = `
<div
  className="albums-detail-wrapper"
>
  <span
    className="go-back"
    onClick={[Function]}
  >
    Go Back
  </span>
  <div
    className="albums-detail"
  >
    <img
      alt="title 1"
      src="image-one"
    />
    <div
      className="text-container"
    >
      <h1>
        title 1
      </h1>
      <h2>
        By 
        name 1
      </h2>
      <p>
        Number 
        1
         on iTunes
      </p>
      <p>
        Released in 
        release-date
      </p>
      <p>
        8
         songs
      </p>
      <p>
        Price in iTunes: 
        price-one
      </p>
      <p>
        rights-data
      </p>
    </div>
  </div>
</div>
`;
